src/lib/fibers/sched.lua

```diff
@@ -92,12 +92,18 @@ function Scheduler:main()
 end
 
 function Scheduler:shutdown()
```

Comment:
Hi, really sorry for the nit-picking here, but I think this code should be in ptree.  The issue is that the scheduler doesn't always have a notion of time, but when it does, the time to which to advance should be passed as the argument to run; but here in sched.lua we don't really know what to do.  Anyway.  I suggest that ptree should call the old `:shutdown()` in a loop until it returns true or 0.1s have passed.  WDYT?